### ISSUE TITLE
Bring back Amazon Video support

### ISSUE THREAD

**Issue by arman86** [mikesteele/dual-captions#80]
I saw this note when I returned to Amazon to watch Diablo Gaurdian:
 > Sorry, Amazon Video support has been removed in v1.6.0.
 >
 >Please open an issue if you'd like support added back in v2.

So, that is what I hope I am doing here--opening an issue. 

{BTW, I love this program and find it extremely useful. Can't program but happy to provide financial support if it is helpful.}

**Comment by mikesteele** [comment]
Hey there!

Thanks so much for opening an issue. I don't track my users at all, so I'm not sure how often users use Amazon. Sorry for the disruption!

Have you ever seen an Amazon Video with two subtitle languages available? The main reason I removed support was because it seemed like there was only ever English. (Please comment a link if a video like that exists) 

I might add back Google Translate support in v2. I'm glad to know that there are users that want it.

Mike

**Comment by arman86** [comment]
Yes. Amazon does seem to have the limitation of only English as a subtitle and no closed captions. 

The translation option is not bad at all. However, if it is possible, I'd like to recommend deepl.com as the translation program of choice. I have had much better luck with it than with Google. I'll open up a new issue for this...

**Comment by schpinkeldinkel** [comment]
please can any more light be shed on enabling amazon video support?

**Comment by mikesteele** [comment]
Are there any videos you can find with more than one language of subtitles? Please comment a link if so.

**Comment by schpinkeldinkel** [comment]
amazon video are extremely poor for subs, easy to find people complaining on forums about the service for this reason. perhaps it's possible to add our own .src if we could create a site integration?

thanks so much mikesteele for this project, happy new year = D

**Comment by mikesteele** [comment]
>  perhaps it's possible to add our own .src if we could create a site integration?

That's a really interesting idea. A SRT file you mean? I could imagine the steps to adding support for SRT files would be the following:

* Add an SRT parser to https://github.com/mikesteele/dual-captions/tree/master/site_integrations/shared

* Change the message payload of the `process-caption-request` message to include a "type", so that the Provider could choose a parser based on caption file type. (At the moment, it only uses site, eg. caption files from Netflix are processed with the Netflix parser) (Here's that message being dispatched in the background page <> and here it is being received <>)

* Add a UI to upload an SRT file in the browser_action and `sendMessageToActiveTab` with the `process-caption-request` message, the SRT file and the new SRT type.

* Add an Amazon site integration. It would not need a `parser` because we're providing the SRT file.

And that should work. I'll see if I can get started on it this week.

> thanks so much mikesteele for this project, happy new year = D

Happy New Year to you as well :) If this project is helpful to you, consider starring the repository or watching to get updates when I make a new release.

- Mike
